core/src/test/java/org/apache/calcite/test/catalog/Fixture.java

```diff
@@ -109,16 +109,23 @@
             .build();
     empListType =
         typeFactory.createArrayType(empRecordType, -1);
+
+    // Subclass ObjectSqlType to make nested field queryable.
     addressType =
-        new ObjectSqlType(SqlTypeName.STRUCTURED,
+        new ObjectSqlType(
+            SqlTypeName.STRUCTURED,
             new SqlIdentifier("ADDRESS", SqlParserPos.ZERO),
             false,
             Arrays.asList(
                 new RelDataTypeFieldImpl("STREET", 0, varchar20Type),
                 new RelDataTypeFieldImpl("CITY", 1, varchar20Type),
                 new RelDataTypeFieldImpl("ZIP", 2, intType),
                 new RelDataTypeFieldImpl("STATE", 3, varchar20Type)),
-            RelDataTypeComparability.NONE);
+            RelDataTypeComparability.NONE) {
+          @Override public StructKind getStructKind() {
+            return StructKind.PEEK_FIELDS_DEFAULT;
```

Comment:
I guess before it was StructKind.FULLY_QUALIFIED, wasn't this enough to demonstrate the problem?

Reply:
Sorry that it was not enough to do that. It will make the nested column unrecognizable by part of the flow. In my personal project I used PEEK_FIELDS_DEFAULT, so I stick to it here.